### ISSUE TITLE
Add customization for using the return form of arrow functions and FC<Props> typing

### ISSUE THREAD

**Issue by etler** [joao-mbn/react-extract#6]
At our company we follow a slightly different formatting style. Currently, the extension produces components like:

```ts
const Component = ({}: Props) => (<InnerComponent/>)
```

We do things slightly differently like:

```ts
const Component: React.FC<Props> = ({}) => {
  return <InnerComponent/>
}
```

It would be a nice convenience to be able to configure the extension to get that output.

By the way, I tried like 3 or 4 other react refactor extensions and this was the best by far!

**Comment by joao-mbn** [comment]
Hey @etler, appreciate the positive feedback. The requested feature is a reasonable addition to the extension and has been added in the #7. This feature is now available at v0.8.0.

Some notes here:

1. If the component has no props and the config "Declare With React FC" is true, it will still declare with React.FC:

```typescript

// this:
const Component: React.FC = () => {
   //...
}

// instead of this:
const Component: = () => {
   //...
}
```

2. Parenthesis will be wrapping the component even in explicit return statements.

This extension avoids having opinions on linting choices and the option of wrapping return statements or not can be configured at your linter. 

I left them by default because I don't format the output component and leave that to the linter. Without the parenthesis, if there's a line break between the return keyword and the output component, the linter might understand it as an early return and cap the rest, like this:

```typescript

// Extension's Output:
const ExtractedComponent = () => {
    return 
    
    <InnerComponent />
}

// After linting:
const ExtractedComponent = () => {
    return;
}

// Instead of this:
const ExtractedComponent = () => {
    return <InnerComponent />;
}

```

**Comment by etler** [comment]
Thanks a bunch! That was really fast. Your notes make sense and I agree with going for explicitness with the return value.

I tried it out and the React.FC works great, although I couldn't get the explicit return to work for some reason. Also, I notice that the true false values were set to strings instead of boolean primitives, and I wasn't sure if that was intentional, but that's not a big deal either way.

**Comment by joao-mbn** [comment]
I made a mistake while getting the config key for explicit return types. It has been addressed at 879f67dbb3b1a7e19bd03e6795e41fabbd35fb87 and should be working on v0.8.1. 

I added to my bucket list implementing a better test suite for the method of picking up the configs.

As to the true/false strings, yes, they were on purpose. Having them as strings, allows me to give examples of what the configuration will do per option. I haven't found a way to do that with booleans. See the images below.

![Screenshot from 2024-05-18 23-11-55](https://github.com/joao-mbn/react-extract/assets/80978135/1618cb06-e7ce-4860-a400-95f6fb88b8ef)

![Screenshot from 2024-05-18 23-12-42](https://github.com/joao-mbn/react-extract/assets/80978135/067e1bfe-f605-4ada-aafe-7d90d8647b04)
